internal/graph/graph.go

```diff
@@ -52,120 +50,101 @@ type ResolutionMetadata struct {
-// relationships.
-type RelationshipIngressType int
+type RelationshipEdgeType int
 
 const (
-	// DirectIngress defines a direct ingress connection between a source object reference
+	// DirectEdge defines a direct connection between a source object reference
 	// and some target user reference.
-	DirectIngress RelationshipIngressType = iota
-	TupleToUsersetIngress
-	ComputedUsersetIngress
+	DirectEdge RelationshipEdgeType = iota
+	// TupleToUsersetEdge defines a connection between a source object reference
+	// and some target user reference that is conditional to a third user reference.
+	TupleToUsersetEdge
+	// ComputedUsersetEdge defines a direct connection between a source object reference
+	// and some target user reference. The difference with DirectEdge is that DirectEdge will involve
+	// a read of tuples and this one will not.
+	ComputedUsersetEdge
 )
 
-func (r RelationshipIngressType) String() string {
+func (r RelationshipEdgeType) String() string {
 	switch r {
-	case DirectIngress:
+	case DirectEdge:
 		return "direct"
-	case ComputedUsersetIngress:
+	case ComputedUsersetEdge:
 		return "computed_userset"
-	case TupleToUsersetIngress:
+	case TupleToUsersetEdge:
 		return "ttu"
 	default:
 		return "undefined"
 	}
 }
 
-type IngressCondition int
+type EdgeCondition int
 
 const (
 
-	// RequiresFurtherEvalCondition indicates an ingress condition whereby results expanded
-	// under such an ingress require further Check evaluation before a determination of the
-	// outcome can be made.
+	// RequiresFurtherEvalCondition indicates an edge condition whereby results found with ReverseExpandQuery
+	// require further Check evaluation before a determination of the outcome can be made.
 	//
 	// Relationships involving intersection ('and') and/or exclusion ('but not') fall under
-	// ingresses with this condition.
-	RequiresFurtherEvalCondition IngressCondition = iota
+	// edges with this condition.
+	RequiresFurtherEvalCondition EdgeCondition = iota
 
-	// NoFurtherEvalCondition indicates an ingress condition whereby results are factual and
+	// NoFurtherEvalCondition indicates an edge condition whereby results found with ReverseExpandQuery are factual and
 	// known to be true and require no further evaluation before a determination of the outcome
 	// can be made.
 	NoFurtherEvalCondition
 )
 
-// RelationshipIngress represents a possible ingress point between some source object reference
-// and a target user reference.
-type RelationshipIngress struct {
+// RelationshipEdge represents a possible relationship some source object reference
+// and a target user reference. The possibility is realized depending on the tuples and on the edge's type.
+type RelationshipEdge struct {
+	Type RelationshipEdgeType
 
-	// The type of the relationship ingress
-	Type RelationshipIngressType
+	// The edge is directed towards this node, which can be like group:*, or group, or group:member
+	TailUsersetNode *openfgav1.RelationReference
```

Comment:
Let's find another name for this. "Tail" is ambiguous here.

This is formally the underlying reference to the target from the incoming "edge". So I may suggest simply calling this the `TargetRelationReference` or `TargetReference`

Reply:
> "Tail" is ambiguous here.

How? You have an edge that starts at A and points at B. Tail is B.

But i can rename